src/java.base/share/classes/jdk/internal/classfile/impl/ClassRemapperImpl.java

```diff
@@ -309,7 +309,7 @@ <S extends Signature> S mapSignature(S signature) {
             case Signature.ClassTypeSig cts ->
                 Signature.ClassTypeSig.of(
                         cts.outerType().map(this::mapSignature).orElse(null),
-                        map(cts.classDesc()),
+                        Util.toInternalName(map(cts.classDesc())), // wrong, needs fix with InnerClasses
```

Comment:
Signatures of inner classes should be disambiguated to avoid this kind of confusions.

Reply:
I aim to preserve the existing behavior for now. We can create a dedicated RFE to disambiguate for both here and InnerClasses later.